src/Console/Output/Progress/ProgressOutputFactory.php

```diff
@@ -20,6 +20,10 @@ final class ProgressOutputFactory
 {
     public static function create(string $outputType, OutputContext $context): ProgressOutputInterface
     {
+        if (self::isValidCustomFactory($outputType)) {
```

Comment:
:-1:

we already have one approach for injecting custom things - custom tokens.
I would not like to have different ways to inject different custom classes.

Reply:
I thought it would be easier for @nunomaduro and other people to use Fixer as a supporting tool, because with custom progress output it would be possible to react to each file's fixing status in a custom way. Some kind of proxy between Fixer's engine and outside world. If you don't like the idea, I will remove it from here when #6848 gets merged and I go back to this one.

Reply:
by no means I said that we should not allow for custom reporter. I said that I'm against proposed implementation of it, as it introduces yet another approach of injecting custom implementation. I do not want to have multiple ways of doing it.

Reply:
but agree about concluding on #6848 before touching this PR review

Reply:
@keradus recently I found out that PHPUnit uses similar approach in its [`--printer` option](https://docs.phpunit.de/en/9.5/textui.html):

> `--printer`
> 
> Specifies the result printer to use. The printer class must extend `PHPUnit\Util\Printer` and implement the `PHPUnit\Framework\TestListener` interface.

Maybe we could reconsider this approach?

Reply:
@SpacePossum did you see the actual implementation? I don't know where you see the problem with streaming data. Main process just calls `onFixerFileProcessed()` on registered printer and at this point our responsibility ends. Users are allowed to do whatever with obtained `FixerFileProcessedEvent` object.

Reply:
I don't think so, signature and public API of `FixerFileProcessedEvent` must remain the same, but new status is _not_ a BC break in my opinion. It's up to consumer to handle statuses. Anyway, I can split this PR (again) and leave `detail` and `meaningful` outputs here and move support for custom outputs to separate PR so we can focus only on new built-in progress outputs here 🙂.

Reply:
> but new status is not a BC break in my opinion

Think that is a big part of it.
Lets say I change the token stream by adding more custom tokens (love to do it btw.,) this has always been considered a BC break because the consuming party doesn't know/expect these token _values_. Same here, is the BC promise on the status on any of the now known values of the _int) or just the _int_?
So if not, if I change the value for `STATUS_INVALID` from 1 to 2 (currently `STATUS_SKIPPED`) it is still an `int`, so we would still honor the interface, but is a BC break?
So yeah I've seen the implementation, but I have yet to see the BC promise defined by the team on the change proposed :)

Reply:
It's the same contract as HTTP status codes in `symfony/http-foundation` - technically each const can change value, but it won't happen. New status codes can be added if specification is extended and it won't be a BC break - consumers just need to align. If anyone decides to use custom printer, takes responsibility of handling statuses and keeping track of upstream set of available values. BC promise is that codes won't be removed or changed, only new may be introduced. I don't see any problem here and this is different than `Tokens` because it's only for consuming, not for manipulating.

Reply:
>consumers just need to align

if only :D

>I don't see any problem here and this is different than Tokens because it's only for consuming, not for manipulating.

Think my point was that is not the case as it is about offering different tokens on patch/minor releases, so not about manipulating tokens by 3rd party. but sure this is not about the tokens stream so maybe not a fair compare from my side.
Think we see different about the cost-value on this, which is fine, just wanted to share my thoughts on it :+1:  :)

Reply:
Support for custom fixers was removed from this PR and will be provided in separate one.